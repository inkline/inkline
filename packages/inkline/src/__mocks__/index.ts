export * from '@inkline/inkline/__mocks__/globalPropertiesMock';
export * from '@inkline/inkline/__mocks__/localStorageMock';
export * from '@inkline/inkline/__mocks__/matchMediaMock';
