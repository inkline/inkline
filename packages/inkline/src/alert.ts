export * from '@inkline/component-alert';
