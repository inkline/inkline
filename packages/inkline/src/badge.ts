export * from '@inkline/component-badge';