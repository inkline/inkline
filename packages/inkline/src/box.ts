export { Box, GridBox } from '@inkline/component-box';
export * from '@inkline/component-box/theme';
