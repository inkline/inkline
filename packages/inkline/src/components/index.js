export { default as IAlert } from '@inkline/inkline/src/components/IAlert';
export { default as IBadge } from '@inkline/inkline/src/components/IBadge';
export { default as IBreadcrumb } from '@inkline/inkline/src/components/IBreadcrumb';
export { default as IBreadcrumbItem } from '@inkline/inkline/src/components/IBreadcrumbItem';
export { default as IButton } from '@inkline/inkline/src/components/IButton';
export { default as IButtonGroup } from '@inkline/inkline/src/components/IButtonGroup';
export { default as ICard } from '@inkline/inkline/src/components/ICard';
export { default as ICollapsible } from '@inkline/inkline/src/components/ICollapsible';
export { default as ICollapsibleItem } from '@inkline/inkline/src/components/ICollapsibleItem';
export { default as IColumn } from '@inkline/inkline/src/components/IColumn';
export { default as IContainer } from '@inkline/inkline/src/components/IContainer';
export { default as IDatatable } from '@inkline/inkline/src/components/IDatatable';
export { default as IDropdown } from '@inkline/inkline/src/components/IDropdown';
export { default as IDropdownDivider } from '@inkline/inkline/src/components/IDropdownDivider';
export { default as IDropdownItem } from '@inkline/inkline/src/components/IDropdownItem';
export { default as IDropdownMenu } from '@inkline/inkline/src/components/IDropdownMenu';
export { default as IHamburgerMenu } from '@inkline/inkline/src/components/IHamburgerMenu';
export { default as IHeader } from '@inkline/inkline/src/components/IHeader';
export { default as IIcon } from '@inkline/inkline/src/components/IIcon';
export { default as ILayout } from '@inkline/inkline/src/components/ILayout';
export { default as ILayoutAside } from '@inkline/inkline/src/components/ILayoutAside';
export { default as ILayoutContent } from '@inkline/inkline/src/components/ILayoutContent';
export { default as ILayoutFooter } from '@inkline/inkline/src/components/ILayoutFooter';
export { default as ILayoutHeader } from '@inkline/inkline/src/components/ILayoutHeader';
export { default as ILinkable } from '@inkline/inkline/src/components/ILinkable';
export { default as IListGroup } from '@inkline/inkline/src/components/IListGroup';
export { default as IListGroupItem } from '@inkline/inkline/src/components/IListGroupItem';
export { default as ILoader } from '@inkline/inkline/src/components/ILoader';
export { default as IMedia } from '@inkline/inkline/src/components/IMedia';
export { default as IModal } from '@inkline/inkline/src/components/IModal';
export { default as INav } from '@inkline/inkline/src/components/INav';
export { default as INavItem } from '@inkline/inkline/src/components/INavItem';
export { default as INavbar } from '@inkline/inkline/src/components/INavbar';
export { default as INavbarBrand } from '@inkline/inkline/src/components/INavbarBrand';
export { default as INavbarItems } from '@inkline/inkline/src/components/INavbarItems';
export { default as IRow } from '@inkline/inkline/src/components/IRow';
export { default as IPagination } from '@inkline/inkline/src/components/IPagination';
export { default as IPopover } from '@inkline/inkline/src/components/IPopover';
export { default as IProgress } from '@inkline/inkline/src/components/IProgress';
export { default as IProgressBar } from '@inkline/inkline/src/components/IProgressBar';
export { default as ISidebar } from '@inkline/inkline/src/components/ISidebar';
export { default as ITable } from '@inkline/inkline/src/components/ITable';
export { default as ITabs } from '@inkline/inkline/src/components/ITabs';
export { default as ITab } from '@inkline/inkline/src/components/ITab';
export { default as ITooltip } from '@inkline/inkline/src/components/ITooltip';

export { default as ICheckable } from '@inkline/inkline/src/components/ICheckable';
export { default as ICheckableGroup } from '@inkline/inkline/src/components/ICheckableGroup';
export { default as ICheckbox } from '@inkline/inkline/src/components/ICheckbox';
export { default as ICheckboxButton } from '@inkline/inkline/src/components/ICheckboxButton';
export { default as ICheckboxGroup } from '@inkline/inkline/src/components/ICheckboxGroup';
export { default as ICheckboxButtonGroup } from '@inkline/inkline/src/components/ICheckboxButtonGroup';
export { default as IInput } from '@inkline/inkline/src/components/IInput';
export { default as IForm } from '@inkline/inkline/src/components/IForm';
export { default as IFormGroup } from '@inkline/inkline/src/components/IFormGroup';
export { default as IFormLabel } from '@inkline/inkline/src/components/IFormLabel';
export { default as IFormError } from '@inkline/inkline/src/components/IFormError';
export { default as IInputNumber } from '@inkline/inkline/src/components/IInputNumber';
export { default as IRadio } from '@inkline/inkline/src/components/IRadio';
export { default as IRadioButton } from '@inkline/inkline/src/components/IRadioButton';
export { default as IRadioGroup } from '@inkline/inkline/src/components/IRadioGroup';
export { default as IRadioButtonGroup } from '@inkline/inkline/src/components/IRadioButtonGroup';
export { default as ISelect } from '@inkline/inkline/src/components/ISelect';
export { default as ISelectOption } from '@inkline/inkline/src/components/ISelectOption';
export { default as ITextarea } from '@inkline/inkline/src/components/ITextarea';
export { default as IToggle } from '@inkline/inkline/src/components/IToggle';
