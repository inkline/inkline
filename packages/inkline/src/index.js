export * from '@inkline/inkline/src/plugin';
export * from '@inkline/inkline/src/components';
