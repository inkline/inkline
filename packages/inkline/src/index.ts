export * from '@inkline/i18n';
export * from '@inkline/utils';
export * from '@inkline/validation';
export * from '@inkline/vue';

export * from './components';
