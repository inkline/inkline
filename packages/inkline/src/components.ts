import type { Component } from 'vue';
import { Alert } from './alert';
import { Badge } from './badge';
import { Breadcrumb, BreadcrumbItem } from './breadcrumb';
import { Button } from './button';
import { ButtonGroup } from './button-group';
import { Card } from './card';
import { Checkbox, CheckboxGroup } from './checkbox';
import { Drawer } from './drawer';
import { FormGroup } from './form-group';
import { Expand } from './expand';
import { Grid } from './grid';
import { Icon } from './icon';
import { Input, Textarea } from './input';
import { Loader } from './loader';
import { Media } from './media';
import { Modal, ModalContainer } from './modal';
import { Nav, NavItem } from './nav';
import { Popover } from './popover';
import { Popup } from './popup';
import { Radio, RadioGroup } from './radio';
import { Toast, ToastContainer } from './toast';
import { Toggle } from './toggle';
import { Tooltip } from './tooltip';
import { Blockquote, Typography } from './typography';

export const components: Record<string, Component> = {
    Alert,
    Badge,
    Blockquote,
    Breadcrumb,
    BreadcrumbItem,
    Button,
    ButtonGroup,
    Card,
    Checkbox,
    CheckboxGroup,
    Drawer,
    FormGroup,
    Expand,
    Grid,
    Icon,
    Input,
    Loader,
    Media,
    Modal,
    ModalContainer,
    Nav,
    NavItem,
    Popover,
    Popup,
    Radio,
    RadioGroup,
    Textarea,
    Toast,
    ToastContainer,
    Toggle,
    Tooltip,
    Typography
};
