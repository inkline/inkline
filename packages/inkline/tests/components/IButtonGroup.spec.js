import { shallowMount } from '@vue/test-utils';
import { IButtonGroup } from '@inkline/inkline/src/components';

describe('Components', () => {
    describe('IButtonGroup', () => {
        let wrapper;

        beforeEach(() => {
            wrapper = shallowMount(IButtonGroup, {
                methods: {
                    created: IButtonGroup.created
                }
            });
        });

        it('should be named correctly', () => {
            expect(IButtonGroup.name).toEqual('IButtonGroup');
        });

        it('should render correctly', () => {
            expect(wrapper.html()).toMatchSnapshot();
        });

        describe('props', () => {
            describe('vertical', () => {
                it('should be defined', () => {
                    expect(wrapper.vm.vertical).toBeDefined();
                    expect(wrapper.vm.vertical).toEqual(false);
                });
            });
            describe('block', () => {
                it('should be defined', () => {
                    expect(wrapper.vm.block).toBeDefined();
                    expect(wrapper.vm.block).toEqual(false);
                });
            });
        });

        describe('created()', () => {
            it('should add class rules to classes provider', () => {
                const classRulesLength = wrapper.vm.classesProvider.length;

                wrapper.vm.created();
                expect(wrapper.vm.classesProvider.length).toEqual(classRulesLength + 1)
            });

            it('should add "-vertical" class if "vertical" property is true', () => {
                const rule = wrapper.vm.classesProvider[wrapper.vm.classesProvider.length - 1];

                expect(rule()).toEqual(expect.objectContaining({ '-vertical': false }));
                wrapper.setProps({ vertical: true });
                expect(rule()).toEqual(expect.objectContaining({ '-vertical': true }));
            });

            it('should add "-block" class if "block" property is true', () => {
                const rule = wrapper.vm.classesProvider[wrapper.vm.classesProvider.length - 1];

                expect(rule()).toEqual(expect.objectContaining({ '-block': false }));
                wrapper.setProps({ block: true });
                expect(rule()).toEqual(expect.objectContaining({ '-block': true }));
            });
        });
    });
});
