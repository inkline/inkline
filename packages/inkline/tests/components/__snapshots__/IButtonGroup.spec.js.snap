// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components IButtonGroup should render correctly 1`] = `<div role="group" class="button-group"></div>`;
